clang/include/module.modulemap

```diff
@@ -111,7 +111,6 @@ module Clang_Diagnostics {
   module All { header "clang/Basic/AllDiagnostics.h" export * }
   module Analysis { textual header "clang/Analysis/Analyses/UnsafeBufferUsageGadgets.def" }
   module AST { header "clang/AST/ASTDiagnostic.h" export * }
-  module Comment { header "clang/AST/CommentDiagnostic.h" export * }
```

Comment:
Do we want to replace this with a reference to `DiagnosticComment.h` instead of removing it entirely?

Reply:
CC @ChuanqiXu9 @Bigcheese

Reply:
I am guessing no, but I admit I am no modulemap expert though.

`module.modulemap` says:

```
module Clang_Diagnostics {
  requires cplusplus

  module All { header "clang/Basic/AllDiagnostics.h" export * }
```

Now, `AllDiagnostics.h` includes `clang/Basic/DiagnosticComment.h`, so `module All` above seems to imply `DiagnosticComment.h`.